### ISSUE TITLE
Fix autotools warnings/errors

### ISSUE THREAD

**Issue by josch** [bestouff/genext2fs#12]
Running `autoreconf -i` produces the following errors (and a few warnings):

```
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
automake: warning: autoconf input should be named 'configure.ac', not 'configure.in'
configure.in:10: warning: AM_INIT_AUTOMAKE: two- and three-arguments forms are deprecated.  For more info, see:
configure.in:10: https://www.gnu.org/software/automake/manual/automake.html#Modernize-AM_005fINIT_005fAUTOMAKE-invocation
configure.in:13: installing './compile'
configure.in:10: installing './install-sh'
configure.in:10: installing './missing'
Makefile.am: error: required file './README' not found
Makefile.am: error: required file './ChangeLog' not found
Makefile.am: installing './depcomp'
automake: warning: autoconf input should be named 'configure.ac', not 'configure.in'
parallel-tests: installing './test-driver'
autoreconf: automake failed with exit status: 1
```

**Comment by josch** [comment]
The files `README` and `ChangeLog` were removed in commits 14a35895f8646c2e2f8ad47712f41cd50fd89f6c and 9436b22a64679f3202d2babdd6b8cfb10fe09be2, respectively. `AM_INIT_AUTOMAKE([foreign])` could be used in `configure.in` to fix the errors.

**Comment by josch** [comment]
The following patch removes the errors and some warnings. I made the changes according to the link in my initial post:

```patch
--- a/configure.in
+++ b/configure.in
@@ -2,12 +2,13 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.59)
-AC_INIT(genext2fs.c)
+AC_INIT([genext2fs], [1.4.2])
+AC_CONFIG_SRCDIR([genext2fs.c])
 
 builtin(include, [m4/ac_func_snprintf.m4])dnl
 builtin(include, [m4/ac_func_scanf_can_malloc.m4])dnl
 
-AM_INIT_AUTOMAKE(genext2fs,1.4.2)
+AM_INIT_AUTOMAKE([foreign])
 AC_CONFIG_HEADER([config.h])
 
 AC_GNU_SOURCE
```

**Comment by Jongy** [comment]
`touch README ChangeLog` also "fixes" this problem :dagger:

**Comment by bestouff** [comment]
Frankly I didn't "autotoolize" genext2fs myself, and I know next to nothing about that. I can't even build genext2fs anymore on my system (Debian testing).  
I'll test this patch, but if it still doesn't work for me I'm in the mood of getting rid of autotools entirely :)

**Comment by josch** [comment]
Current genext2fs builds just fine on Debian testing (I'm the current maintainer of genext2fs in Debian). If you post an error, maybe I can help?

But if you want to get rid of autotools anyways, you will not hear a complaint from me. ^^

**Comment by bestouff** [comment]
Well, for a start from a fresh `git clone`:
```
./autogen.sh: 13: Bad substitution
```

**Comment by Jongy** [comment]
> ./autogen.sh: 13: Bad substitution

Try with `bash`, I assume `autogen.sh` uses bash-only shell substitutions (and its interpreter is set to `/bin/sh`, might change that as well)

**Comment by bestouff** [comment]
```
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
automake: warning: autoconf input should be named 'configure.ac', not 'configure.in'
configure.in:10: warning: AM_INIT_AUTOMAKE: two- and three-arguments forms are deprecated.  For more info, see:
configure.in:10: https://www.gnu.org/software/automake/manual/automake.html#Modernize-AM_005fINIT_005fAUTOMAKE-invocation
configure.in:13: installing './compile'
configure.in:10: installing './install-sh'
configure.in:10: installing './missing'
Makefile.am: error: required file './README' not found
Makefile.am: error: required file './ChangeLog' not found
Makefile.am: installing './depcomp'
automake: warning: autoconf input should be named 'configure.ac', not 'configure.in'
parallel-tests: installing './test-driver'
*** automake failed :(
```

**Comment by Jongy** [comment]
Now `touch README ChangeLog` will get rid of the "required file" errors

**Comment by bestouff** [comment]
All right, I have applied @josch's patch and changed `autogen.sh`'s interpreter in 61df15addc71f08baafc6243f857fe17fe2c43a6 - things look better, thanks !

**Comment by bestouff** [comment]
Should I rename `configure.in` to `configure.ac` to get rid of the warning ?

**Comment by josch** [comment]
I think this is because `cache.h` is missing from `EXTRA_DIST` in `Makefile.am`.

**Comment by bestouff** [comment]
Yes, that and a few other problems, fixed in fc74406677f9e9167b13121f7e3b988ca6ef6e57. Thanks @josch.
